### ISSUE TITLE
Reading a csv file as arrayBuffer instead of text in browser

### ISSUE THREAD

**Issue by d3bgger** [finos/perspective#2746]
### Description of Problem:
If I understand correctly, in the browser, the only way to pass a csv file to `worker.table()` is by text.
So from a file selected by the user we need `let data = await file.text();`. 

This is inefficient and fails for large files.

### Potential Solutions:

Can there be support for a buffer to be passed directly for csv files?  For example:
```
let fileBuffer = await file.arrayBuffer();
let data = new Uint8Array(fileBuffer);

const table = await worker.table(data);
```

**Comment by texodus** [comment]
> This is inefficient and fails for large files.

We could in theory add this, but I don't expect the change to impact perspective performance ("efficiency") or max data size. In your example, there is (tmk) virtually no difference between calling `file.arrayBuffer()` and `file.text()` wrt aggregate memory allocation in the, unless you need the `ArrayBuffer` for something else. In both cases, the large string-or-arraybuffer is copied into Perspective when we load it.

Regardless, to fix a specific incompatibility with a file (size related or otherwise), we'd need an issue that repros the actual failure to comment on a fix. We do have a few paths for increasing Perspective WASM build max data size, some API improvements to minimize _internal_ WASM heap copying, and maybe a future move to `Memory64` support when it is available.

**Comment by d3bgger** [comment]
Hi, thanks for taking the time. So here's a minimal test:

```
let startTime = performance.now();
let dataTxt = await file.text();
var endTime = performance.now();
console.log(`txt elapsed ${parseInt(endTime - startTime)}ms`);
console.log(dataTxt.length);

startTime = performance.now();
let fileBuffer = await file.arrayBuffer();
let data = new Uint8Array(fileBuffer);
endTime = performance.now();
console.log(`buffer elapsed ${parseInt(endTime - startTime)}ms`);
console.log(data.length);
```

output for 200MB file:
```
txt elapsed 532ms
202925232
buffer elapsed 138ms
202925232
```

output for 900MB file:
```
txt elapsed 3158ms
0
buffer elapsed 520ms
914988614
```

**Comment by texodus** [comment]
Hrm, ok .. interesting. I am still not convinced this will be loadable in Perspective WASM (see below). But, adding this support is not difficult, so we can start with this. `Blob` or `Stream` support may be better than `ArrayBuffer` in this regard though, the latte may engine support to fully utilize.

900mb is pushing the limits of what I expect Perspective to be able to accomplish in its current form purely in the browser (`perspective-python` e.g. should have no problem with this). Data size vs memory is complicated as it depends highly on the data schema and composition, but the ultimate limitation is just wasm32. Depending on your browser/platform/version (this is in the Issue template for filing bugs btw), you may not be able to reliably allocate 2gb of total heap memory in the browser in this mode, and if you do you may not be able to e.g. create a large/dense pivot on this without OOM-ing.

 In the future, [Memory64](https://github.com/WebAssembly/proposals) is designed to address this.

**Comment by 8296368008** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>CSV to ArrayBuffer</title>
</head>
<body>
    <h1>Upload CSV and Read as ArrayBuffer</h1>
    <input type="file" id="csvFileInput" accept=".csv">
    <pre id="output"></pre>

    <script>
        document.getElementById('csvFileInput').addEventListener('change', function(event) {
            const file = event.target.files[0];
            if (file) {
                const reader = new FileReader();
                
                // Read the file as ArrayBuffer
                reader.readAsArrayBuffer(file);

                reader.onload = function(e) {
                    const arrayBuffer = e.target.result;
                    
                    // Display ArrayBuffer info in the output
                    document.getElementById('output').textContent = `ArrayBuffer Length: ${arrayBuffer.byteLength}`;
                    
                    // Optionally, you can convert the ArrayBuffer to a typed array or process the CSV file further
                    const uint8Array = new Uint8Array(arrayBuffer);
                    console.log(uint8Array); // Log to check raw data
                };

                reader.onerror = function() {
                    console.error("An error occurred while reading the file.");
                };
            }
        });
    </script>
</body>
</html>

**Comment by texodus** [comment]
This is available in `3.1.0`, use the `format` option when creating a table:

```javascript
data = '{"x": [1,2,3], "y": [4,5,6]}';
table = await client.table(data, {format: "columns"});
```
